app/src/main/java/org/groundplatform/android/util/CoordinatesExt.kt

```diff
@@ -37,3 +38,23 @@ private fun decimalToDms(latOrLong: Double): String {
   val (degrees, minutes, seconds) = dmsFormat.split(":".toRegex())
   return "$degrees°$minutes'$seconds\""
 }
+
+fun List<Coordinates>.tooltipDistanceIfLineStringClosed(): Double? {
+  if (size < 2 || isClosedLineString()) return null
+  return this[size - 2].distanceTo(this[size - 1])
+}
+
+fun List<Coordinates>.midPointToLastSegment(): LatLng? {
+  if (size < 2 || isClosedLineString()) return null
+  val end = this[size - 1]
+  val start = this[size - 2]
+  return LatLng((start.lat + end.lat) / 2, (start.lng + end.lng) / 2)
+}
+
+private fun Coordinates.distanceTo(other: Coordinates): Double {
+  val result = FloatArray(1)
+  Location.distanceBetween(this.lat, this.lng, other.lat, other.lng, result)
+  return result[0].toDouble()
+}
+
+private fun List<Coordinates>.isClosedLineString(): Boolean = size >= 4 && first() == last()
```

Comment:
This doesn't actually test whether these coords are part of a `LineString`, which is a semantic distinction it cannot know about.  I think `isClosed` would be more correct here, in which case `size` must be `>0`.

We shouldn't encode the UI-specific logic (only show when label size >=4) here, but rather in the business layer.

In Koltin you can remove the return type, since it's immediate clear this is a Boolean expression.